python/src/uagents/agent.py

```diff
@@ -1020,6 +1059,14 @@ async def _process_message_queue(self):
                     )
                     continue
 
+            if state and state != self.state:
+                await _send_error_message(
+                    context,
+                    sender,
+                    ErrorMessage(error="Agent not in correct state to handle message"),
```

Comment:
Should we support custom error messages the developer can provide with the decorator?

Reply:
Currently if an agent has no handler for a message, it is simply ignored, so I kind of think we should treat being in wrong state to handle a message the same as not having a handler for that message. So for consistency with the current setup, we would ignore that case. However, I can see the case for sending error messages back in both cases, because at the moment I don't think the sending agent gets any feedback when sending an invalid or incorrect message to an agent.